--- openconnect-3.02/Makefile.orig	2011-05-04 18:53:39.109630396 +0200
+++ openconnect-3.02/Makefile	2011-05-04 17:45:05.059718203 +0200
@@ -15,23 +15,23 @@ endif
 # dir; there's no need to install it anywhere (we link it statically).
 ifdef OPENSSL
 SSL_CFLAGS += -I$(OPENSSL)/include
-SSL_LDFLAGS += $(OPENSSL)/libssl.a $(OPENSSL)/libcrypto.a
+SSL_LIBS += $(OPENSSL)/libssl.a $(OPENSSL)/libcrypto.a
 else
 SSL_CFLAGS += $(shell pkg-config --cflags libssl)
-SSL_LDFLAGS += $(shell pkg-config --libs libssl)
-ifeq ($(SSL_LDFLAGS),)
+SSL_LIBS += $(shell pkg-config --libs libssl)
+ifeq ($(SSL_LIBS),)
 $(error "No OpenSSL support. Cannot continue");
 endif
 endif
 
 XML2_CFLAGS += $(shell xml2-config --cflags) 
-XML2_LDFLAGS += $(shell xml2-config --libs)
-ifeq ($(XML2_LDFLAGS),)
+XML2_LIBS += $(shell xml2-config --libs)
+ifeq ($(XML2_LIBS),)
 $(error "No libxml2 support. Cannot continue");
 endif
 
 CFLAGS := $(OPT_FLAGS) $(SSL_CFLAGS) $(XML2_CFLAGS) $(EXTRA_CFLAGS)
-LDFLAGS := -lz $(SSL_LDFLAGS) $(XML2_LDFLAGS) $(EXTRA_LDFLAGS)
+LIBS := -lz $(SSL_LIBS) $(XML2_LIBS) $(EXTRA_LIBS)
 
 ifdef SSL_UI
 CFLAGS += -DSSL_UI
@@ -45,7 +45,7 @@ endif
 
 ifneq ($(LIBPROXY_HDR),)
 CFLAGS += -DOPENCONNECT_LIBPROXY -DLIBPROXY_HDR=\"$(LIBPROXY_HDR)\"
-LDFLAGS += -lproxy
+LIBS += -lproxy
 endif
 
 OPENCONNECT_OBJS := xml.o main.o $(SSL_UI)
@@ -68,7 +68,7 @@ version.c: $(patsubst %.o,%.c,$(VERSION_
 	@./version.sh
 
 openconnect: $(OPENCONNECT_OBJS) $(CONNECTION_OBJS) libopenconnect.a
-	$(CC) -o $@ $^ $(LDFLAGS)
+	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
 
 %.o: %.c
 	$(CC) -c -o $@ $(CFLAGS) $(CFLAGS_$@) $< -MD -MF .$@.dep
