--- openconnect-5.00/configure.ac.orig	2013-05-15 22:14:32.000000000 +0200
+++ openconnect-5.00/configure.ac	2013-05-25 20:20:27.607437685 +0200
@@ -326,18 +326,6 @@
     AC_CHECK_FUNC(gnutls_pkcs11_add_provider,
 		 [PKG_CHECK_MODULES(P11KIT, p11-kit-1, [AC_DEFINE(HAVE_P11KIT)
     			      		  AC_SUBST(P11KIT_PC, p11-kit-1)], [:])], [])
-    LIBS="$oldlibs -ltspi"
-    AC_MSG_CHECKING([for tss library])
-    AC_LINK_IFELSE([AC_LANG_PROGRAM([
-		   #include <trousers/tss.h>
-		   #include <trousers/trousers.h>],[
-		   int err = Tspi_Context_Create((void *)0);
-		   Trspi_Error_String(err);])],
-		  [AC_MSG_RESULT(yes)
-		   AC_SUBST([TSS_LIBS], [-ltspi])
-		   AC_SUBST([TSS_CFLAGS], [])
-		   AC_DEFINE(HAVE_TROUSERS, 1)],
-		  [AC_MSG_RESULT(no)])
     LIBS="$oldlibs"
     CFLAGS="$oldcflags"
 fi
